src/otx/engine/engine.py

```diff
@@ -340,7 +340,7 @@ def test(
         datamodule = datamodule if datamodule is not None else self.datamodule
 
         is_ir_ckpt = Path(str(checkpoint)).suffix in [".xml", ".onnx"]
-        if is_ir_ckpt and not isinstance(model, OVModel):
+        if is_ir_ckpt:
```

Comment:
The only case where `OVModel` is already used is when using `openvino_model.yaml` as config file, as far as I know. In this case, I don't think we need to update the `datamodule` pipeline and `model`. Is there any reason why this syntax should disappear?

Reply:
@harimkang  `datamodule` is set with `OTXTileTestDataset` following `otx test --work_dir /tmp/tile/ --config src/otx/recipe/instance_segmentation/openvino_model.yaml --checkpoint /tmp/tile/.latest/export/exported_model.xml`

As Model API has tiling API, there's no need to use OTXTileTestDataset, thus, we should reset the datamodule if model IR is specified. Tbh, I don't quite understand why there's this condition, and I think it should be changed to `is_ir_ckpt and isinstance(model, OVModel)`

Reply:
This is for the natural flow of the engine. After engine.train and export, in order to use the testing ir model, we need to create a new engine by creating an OVModel. This statement eliminates the need to create a new engine by allowing OVModel to be built and used for each task when the checkpoint is xml.

Reply:
Thanks for the explanation, I made a new update.